py/objarray.c

```diff
@@ -400,6 +400,70 @@ STATIC mp_obj_t array_extend(mp_obj_t self_in, mp_obj_t arg_in) {
 STATIC MP_DEFINE_CONST_FUN_OBJ_2(array_extend_obj, array_extend);
 #endif
 
+#if MICROPY_PY_BUILTINS_BYTEARRAY
+STATIC mp_obj_t buffer_finder(size_t n_args, const mp_obj_t *args, int direction, bool is_index) {
+    mp_check_self(MP_OBJ_IS_TYPE(args[0], &mp_type_bytearray));
+    const mp_obj_type_t *self_type = mp_obj_get_type(args[0]);
+
+    mp_buffer_info_t haystack_bufinfo;
+    mp_get_buffer_raise(args[0], &haystack_bufinfo, MP_BUFFER_READ);
+
+    mp_buffer_info_t needle_bufinfo;
+    mp_get_buffer_raise(args[1], &needle_bufinfo, MP_BUFFER_READ);
+
+    if (mp_binary_get_size('@', needle_bufinfo.typecode, NULL) != 1) {
+        const qstr src_name = mp_obj_get_type(args[1])->name;
+        nlr_raise(mp_obj_new_exception_msg_varg(&mp_type_TypeError,
+            translate("'%q' object is not bytes-like"),
+            src_name));
+    }
```

Comment:
Is it possible for needle to not be bytes-like if `mp_get_buffer_raise()` already succeeded on it?

Reply:
Yes, it could be an array of values that are more than a single byte.